src/s2/s2builderutil_lax_polygon_layer_test.cc

```diff
@@ -52,6 +53,8 @@ string ToString(DegenerateBoundaries degenerate_boundaries) {
     case DegenerateBoundaries::DISCARD_HOLES: return "DISCARD_HOLES";
     case DegenerateBoundaries::DISCARD_SHELLS: return "DISCARD_SHELLS";
     case DegenerateBoundaries::KEEP: return "KEEP";
+    default:
+      throw std::runtime_error("Unrecognized DegenerateBoundaries value!");
   }
 }
```

Comment:
`s2geometry` doesn't use exceptions, so `return "UNKNOWN";` or `LOG(FATAL) << ...` *after* the switch would be better, but I'm really wondering why you're getting this warning when the cases are exhaustive.

Reply:
I'm also a little mystified as to why the compiler wouldn't detect this, but I do get this warning with an unaltered configuration using G++ 9.3.0. I've updated the commit to use `LOG`, as you suggest.